core/src/main/scala/cats/syntax/either.scala

```diff
@@ -311,10 +315,19 @@ final class CatchOnlyPartiallyApplied[T] private[syntax] {
     }
 }
 
+final class LeftOps[A, B](val left: Left[A, B]) extends AnyVal {
+  /** Cast the right type parameter of the `Left`. */
+  def rightCast[C]: Either[A, C] = left.asInstanceOf[Either[A, C]]
+}
+
+final class RightOps[A, B](val right: Right[A, B]) extends AnyVal {
+  /** Cast the left type parameter of the `Right`. */
+  def leftCast[C]: Either[C, B] = right.asInstanceOf[Either[C, B]]
+}
```

Comment:
I chose to name them like this to follow the `leftMap` convention - the name indicates which type parameter side to cast.